src/control/Control.cpp

```diff
@@ -1623,7 +1623,7 @@ void Control::undoRedoChanged() {
 
 void Control::undoRedoPageChanged(PageRef page) {
     if (!this->changedPages.empty() &&
```

Comment:
You are right about the fix. The check whether `changedPages` is non-empty can be removed then, however. So please remove it.

Reply:
Done